packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -1918,6 +1918,22 @@ void main() {
     expect(selectedIndex, 2);
   });
 
+  testWidgets('onDestinationSelected is not called if null', (WidgetTester tester) async {
+    const int selectedIndex = 0;
+    
```

Comment:
```suggestion

```

Reply:
(this is what the analyzer complains about)

Reply:
Got it! Thanks!